tests/kuberay_utils/utils.py

```diff
@@ -41,29 +41,30 @@ def create_ray_cluster(template_name, ray_version, ray_image):
         yamlfile = template.substitute(
             {'ray_image': ray_image, 'ray_version': ray_version}
         )
-        with tempfile.NamedTemporaryFile('w', delete=False) as ray_cluster_yaml:
-            ray_cluster_yaml.write(yamlfile)
-            context['filepath'] = ray_cluster_yaml.name
-
-    for k8s_object in yaml.safe_load_all(yamlfile):
-        if k8s_object['kind'] == 'RayCluster':
-            context['cr'] = k8s_object
-            break
-
-    try:
-        # Create a RayCluster
-        ray_cluster_add_event = RayClusterAddCREvent(
-            custom_resource_object = context['cr'],
-            rulesets = [],
-            timeout = 90,
-            namespace='default',
-            filepath = context['filepath']
-        )
-        ray_cluster_add_event.trigger()
-        return ray_cluster_add_event
-    except Exception as ex:
-        logger.error(f"RayClusterAddCREvent fails to converge: {str(ex)}")
-    raise Exception("create_ray_cluster fails")
+    with tempfile.NamedTemporaryFile('w', suffix = '_ray_cluster_yaml') as ray_cluster_yaml:
+        ray_cluster_yaml.write(yamlfile)
+        ray_cluster_yaml.flush()
```

Comment:
Would you mind explaining why do we need to use `flush()` in the PR description?

https://discuss.python.org/t/should-tempfile-namedtemporaryfile-automatically-flush-the-buffer/17388